packages/react/src/primitives/Pagination/useRange.ts

```diff
@@ -13,17 +13,17 @@ export const ELLIPSIS = '...';
  * @returns an array that contains the range of numbers to be rendered
  */
 export const useRange = (
-  currentPage: number,
-  totalPages: number,
-  siblingCount = 1
+  currentPageParam: number,
+  totalPagesParam: number,
+  siblingCountParam = 1
 ): (string | number)[] => {
   const range = useMemo(() => {
     // The current page should not be less than 1
-    currentPage = Math.max(currentPage, 1);
+    const currentPage = Math.max(currentPageParam, 1);
```

Comment:
Error fixed here was:

Assignments to the `currentPage` variable from inside React Hook useMemo will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useMemo.

Solved by creating a local variable `currentPage`.